### ISSUE TITLE
JSON query params are initialized with {} when they are not present.

### ISSUE THREAD

**Issue by wertzui** [Nas3nmann/ngx-deep-linking#13]
When a query parameter is of type `'json'` and it is `null` or `undefined`, `{ }` is returned.
https://github.com/Nas3nmann/ngx-deep-linking/blob/090214a6a12694d96c9a246c67e164dbb108a8b7/libs/ngx-deep-linking/src/lib/router-input-wrapper/deep-linking-wrapper.component.ts#L137-L150

This makes it impossible to test the input for `null` or `undefined` and one must apply a workaround like the following one (as long as `{ }` is not a valid value in which case it does not work at all.
```
  _foo?: Foo;

  @Input()
  set foo(value: Foo) {
    if(value && Object.entries(value).length > 0)
      this._foo = value;
  }
  get foo() {
    return this._foo;
  }
```

**Comment by wertzui** [comment]
I'm not talking about a string with the value `'null'` or `'undefined'`. I'm talking about `param` itself being `null` or `undefined`.

My proposal is to change the method as follows
```
 private getTypedQueryParam(deepLinkingParam: DeepLinkingQueryParam, param?: string): string | number | Record<string, unknown> | unknown[] | null | undefined { 
   if (param === undefined || param === null) { 
     return param; 
   } 
  
   switch (deepLinkingParam.type) { 
     case 'string': 
       return param; 
     case 'number': 
       return Number(param); 
     case 'json': 
       return JSON.parse(param); 
   } 
 } 
```

**Comment by Nas3nmann** [comment]
Yes I know, so basically this is how the overall bigger picture should be IMO:

Url query param not present -> input null (regardless of type)

Input null or undefined -> do not include (or even remove) url query param

Agree?

**Comment by wertzui** [comment]
I would not change a missing query param to `null`, but just return it. This should normally result in it being undefined.
However someone might come along with a possibility to set a query param to `null `in which case null should be returned.

For all types:
`undefined `=> `undefined`
`null `=> `null`

For specific types:
(type string) `string` => `string`
(type number) `string` => `number`
(type json) `string` => `unknown`

**Comment by Nas3nmann** [comment]
Yes that sounds good. Feel free to implement it that way.

**Comment by Nas3nmann** [comment]
Fixed in 1.0.8
